### PR TITLE
render Annotation as flag. Merging from develop branch.

### PR DESCRIPTION
**Pull request by cnardi-dev**
see the TODOs in the comments.  know issues:

- x position of annotation is not syncing with x axis (also update on panning/zooming)
- need to calc offset to position for multiple annotations
- text field is too large and can't be used.  need alternate way to inspect/copy description

### CODE REVIEW

**Comment by cnardi-dev** [comment]
screen shot 2016-11-11 at 6 29 42 pm

File: ArgusWeb/app/js/directives/charts/lineChart.js [hunk 1]
```diff
@@ -589,6 +584,42 @@ angular.module('argus.directives.charts.lineChart', [])
...
+            function updateAnnotations() {
+                if (!scope || scope.series.length > 1 ) return;
+
+                var flagSeries = scope.series[0].flagSeries.data
+                var flagsG = d3.select('svg').select('.flags');
+                var label = flagsG.selectAll("flagItem")
+                    .data(flagSeries)
+                    .enter().append("g")
+                    .attr("class", "flagItem")
+                    .attr("transform", function(d) {
```
Comment (xizi-xu) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: This is just a hard code value right?

Comment (cnardi-dev) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: Not harding the data, its making an API call specifically for the Annotations, which adds 'flagSeries', to the 'series' object before the lineChart directive is called.

Comment (cnardi-dev) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: flagseries-data

Comment (cnardi-dev) [inline ArgusWeb/app/js/directives/charts/lineChart.js]: If there is NO flagSeries data, then this object will not be present.

**Pull request merged**